kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/BigQuerySinkConnectorIT.java

```diff
@@ -141,86 +147,112 @@
         "forty-two",
         boxByteArray(new byte[] { 0x0, 0xf, 0x1E, 0x2D, 0x3C, 0x4B, 0x5A, 0x69, 0x78 })
     ));
-    testCases.put("primitives", expectedPrimitivesRows);
+    result.add(new Object[] {"primitives", expectedPrimitivesRows});
 
     List<List<Object>> expectedLogicalTypesRows = new ArrayList<>();
     expectedLogicalTypesRows.add(Arrays.asList(1L, 0L, 0L));
     expectedLogicalTypesRows.add(Arrays.asList(2L, 42000000000L, 362880000000L));
     expectedLogicalTypesRows.add(Arrays.asList(3L, 1468275102000000L, 1468195200000L));
-    testCases.put("logical-types", expectedLogicalTypesRows);
+    result.add(new Object[] {"logical-types", expectedLogicalTypesRows});
 
-    TEST_CASES = Collections.unmodifiableMap(testCases);
+    return result;
   }
 
   private static final String TEST_CASE_PREFIX = "kcbq_test_";
 
-  private static final Collection<String> TEST_TOPICS = TEST_CASES.keySet().stream()
-      .map(tc -> TEST_CASE_PREFIX + tc)
-      .collect(Collectors.toList());
+  // Share a single embedded Connect and Schema Registry cluster for all test cases to keep the runtime down
+  private static BaseConnectorIT testBase;
+  private static RestApp schemaRegistry;
+  private static String schemaRegistryUrl;
+
+  private final String testCase;
+  private final List<List<Object>> expectedRows;
+  private final String topic;
+  private final String table;
+  private final String connectorName;
 
-  private RestApp restApp;
-  private String schemaRegistryUrl;
   private Producer<byte[], byte[]> valueProducer;
   private int numRecordsProduced;
 
-  @Before
-  public void setup() throws Exception {
-    Collection<String> tables = TEST_TOPICS.stream()
-        .map(this::suffixedAndSanitizedTable)
-        .collect(Collectors.toSet());
-    BucketClearer.clearBucket(keyFile(), project(), gcsBucket(), gcsFolder(), keySource());
-    TableClearer.clearTables(newBigQuery(), dataset(), tables);
-
-    startConnect();
-    restApp = new RestApp(
+  public BigQuerySinkConnectorIT(String testCase, List<List<Object>> expectedRows) {
+    this.testCase = testCase;
+    this.expectedRows = expectedRows;
+
+    this.topic = TEST_CASE_PREFIX + testCase;
+    this.table = testBase.suffixedAndSanitizedTable(topic);
+    this.connectorName = "bigquery-connector-" + testCase;
```

Comment:
Ah now you create a bunch of individual connectors. Does this run the tests serially now instead of in-parallel? Is there a hint that we can give to junit to run the test cases in parallel? is that already set up in the pom?

Reply:
These are run serially; as far as I know it's not possible to make a single test class run all its cases in parallel using only vanilla JUnit 4. The runtime penalty is mitigated by sharing embedded instances across test cases.

I'm happy to upgrade to JUnit 5 as a follow-up item, but I don't think we should block on that for this incremental change, which improves the code base to be more idiomatic in its use of JUnit and provides a better signal of success/failure for each run.

Reply:
Yeah I don't think this is a blocker, worker startup might dominate the test runtime anyway.